fetch_napari_data.py

```diff
@@ -292,68 +263,125 @@ def build_plugins_dataframe() -> pd.DataFrame:
-            )
-            last_updated_date = get_version_release_date(
-                pypi_info, plugin_latest_release
-            )
-
-            # grab pypi project link
-            home_pypi = pypi_info["info"].get("package_url", "")
-            plugin_data.update(
-                {
-                    "created_at": initial_release_date,
-                    "modified_at": last_updated_date,
-                    "home_pypi": home_pypi,
-                }
-            )
-
-        if manifest_info:
-            flatten_and_merge(plugin_data, manifest_info)
-
-        all_plugin_data.append(plugin_data)
-
-    with ThreadPoolExecutor() as executor:
-        executor.map(process_plugin, summary_df)
-
-    return pd.DataFrame(all_plugin_data)
+    )
+
+    widgets = [widget["display_name"] for widget in contributions["widgets"] or []]
+
+    sample_data = [
+        sample["display_name"] for sample in contributions["sample_data"] or []
+    ]
+
+    return PluginPageData(
+        normalized_name=plugin_normalized_name,
+        name=name,
+        display_name=display_name,
+        version=plugin_latest_release,
+        created_at=initial_release_date,
+        modified_at=last_updated_date,
+        authors=authors,
+        author_emails=emails,
+        license=package_license,
+        home_pypi=home_pypi,
+        home_github=home_github,
+        home_other=home_other,
+        summary=plugin_summary_data.get("summary", ""),
+        package_metadata_requires_python=package_metadata.get("requires_python", ""),
+        package_metadata_requires_dist=package_metadata.get("requires_dist", []),
+        package_metadata_description=package_metadata.get("description", ""),
+        package_metadata_classifiers=package_metadata.get("classifiers", []),
+        contributions_readers_filename_patterns=reader_patterns,
+        contributions_writers_filename_extensions=writer_extensions,
+        contributions_widgets=widgets,
+        contributions_sample_data=sample_data,
+    )
+
+
+def create_search_index(plugins: list[PluginPageData], data_dir: str):
+    search_index = [
+        {
+            "name": plugin.name,
+            "display_name": plugin.display_name,
+            "normalized_name": plugin.normalized_name,
+            "summary": plugin.summary,
+            "authors": plugin.authors,
+            "author_emails": plugin.author_emails,
+            "filename_patterns": plugin.contributions_readers_filename_patterns,
+            "filename_extensions": plugin.contributions_writers_filename_extensions,
+            "widgets": plugin.contributions_widgets,
+            "sample_data": plugin.contributions_sample_data,
+            "contributions": get_plugin_types(plugin),
+        }
+        for plugin in plugins
+    ]
+
+    # newline-delimited JSON format for search index
+    # allows for efficient streaming and parsing
+    with open(f"{data_dir}/search_index.ndjson", "w") as file:
```

Comment:
I think before writing out the search index we should sort by the last updated date (I think we call it `modified_at`. If I understand the search correctly, that should ensure that the streamed in search results are also sorted in this manner, right?

Reply:
I think it's already sorted this way, because it's based on `build_plugin_list()` which sorts the data.

Testing myself the plugins appear sorted by (reverse) modified date both in the full list and when searching. Do you (or @TimMonko) have a way to reproduce a wrong ordering?

They should also stay in order because searching now toggles visibility of the HTML elements instead of removing/adding them.